torch/csrc/jit/codegen/cuda/ir_cloner.h

```diff
@@ -0,0 +1,91 @@
+
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/codegen/cuda/dispatch.h>
+
+#include <unordered_map>
+#include <vector>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+class Fusion;
+
+// Clones nodes from an exiting Fusion
+class TORCH_CUDA_API IrCloner : private OptInConstDispatch {
+  friend struct Statement;
+
+ public:
+  explicit IrCloner(Fusion* new_fusion) : fusion_(new_fusion) {}
+
+  Statement* clone(const Statement* statement);
+
+  template <class T>
+  T* clone(const T* node) {
```

Comment:
Sorry, having a hard time understanding what this is used for and where.

Reply:
Convenience + type safety. IrCloner::clone(const Statement* statement) happy clones any Statement* (including derived classes) and returns a Statement*.

W/o this helper template you'd need to type everywhere:

val = ir_cloner->clone(src)->as<Val>()

With it, we can simply say

val = ir_cloner->clone(src)

Reply:
ah okay, thanks!